tests/phpunit/integration/Modules/AdsTest.php

```diff
@@ -321,6 +321,18 @@ public function test_template_redirect__with_pax_flag_and_pax_conversion_id_sett
 			);
 		}
 	}
+	/**
+	 * Test if Ads module correctly registers Ads Measurement connection check filter.
+	 */
+	public function test_ads_module_adds_measurement_connection_check_filter() {
+		$ads = new Ads( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) );
+		$ads->register();
+
+		$checks = apply_filters( 'googlesitekit_ads_measurement_connection_checks', array() );
```

Comment:
This test is essentially a sub-test/variant of `test_register`. It does a bit more than just check the hooks are set though (although that part would be good to add to that test), we should test that callback it adds returns `true` or `false` as we expect. Again, it's important to remove all filters before `register` to ensure we're only testing the callback added by this module.